Flashcards.Dejmenek/Flashcards.Dejmenek/DataAccess/Repositories/StacksRepository.cs

```diff
@@ -0,0 +1,223 @@
+﻿using Dapper;
+using Flashcards.Dejmenek.DataAccess.Interfaces;
+using Flashcards.Dejmenek.Models;
+using System.Configuration;
+using System.Data.SqlClient;
+
+namespace Flashcards.Dejmenek.DataAccess.Repositories
+{
+    public class StacksRepository : IStacksRepository
+    {
+        private static readonly string _connectionString = ConfigurationManager.ConnectionStrings["LocalDbConnection"].ConnectionString;
+        public void AddStack(string name)
+        {
+            using (var connection = new SqlConnection(_connectionString))
+            {
+                string sql = @"INSERT INTO Stacks (Name) VALUES
+                               (@Name)";
+
+                connection.Execute(sql, new
+                {
+                    Name = name
+                });
+            }
+        }
+
+        public void AddFlashcard(int stackId, string front, string back)
```

Comment:
I see you put flashcard operations in the stack repository. Ideally you want to put all related methods in one class. The flashcard methods should move to the flashcard repository 😅

Reply:
I didn't know what to do with it because I had to include an option to add a new flashcard in `ManageStacks` and `ManageFlashcards` section. Should I pass `FlashcardsRepository` to `StacksController` and use `AddFlashcard` method in `AddFlashcardToStack` ?
It would be something like this:
```
private readonly StacksRepository _stacksRepository;
private readonly FlashcardsRepository _flashcardsRepository;
private readonly UserInteractionService _userInteractionService;
public Stack? CurrentStack { get; private set; }

public StacksController(StacksRepository stacksRepository, UserInteractionService userInteractionService, FlashcardsRepository flashcardsRepository)
{
    _stacksRepository = stacksRepository;
    _userInteractionService = userInteractionService;
}

public void AddFlashcardToStack()
{
    string front = _userInteractionService.GetFlashcardFront();
    string back = _userInteractionService.GetFlashcardBack();

    _flashcardsRepository.AddFlashcard(CurrentStack.Id, front, back);
}
```

Reply:
yep :100: you can do that